src/java.desktop/unix/native/common/awt/fontpath.c

```diff
@@ -935,7 +935,7 @@ Java_sun_font_FontConfigManager_getFontConfig
         if (cacheDirs != NULL) {
             while ((cnt < max) && (cacheDir = (*FcStrListNext)(cacheDirs))) {
                 jstr = (*env)->NewStringUTF(env, (const char*)cacheDir);
-                JNU_CHECK_EXCEPTION(env);
+                JNU_CHECK_EXCEPTION_AND_CLEANUP(env, (*FcStrListDone)(cacheDirs));
 
                 (*env)->SetObjectArrayElement(env, cacheDirArray, cnt++, jstr);
```

Comment:
Probably we should add the check+cleanup after the SetObjectArrayElement? Otherwise, we may call NewStringUTF while an exception is raised by the SetObjectArrayElement.

Reply:
??? You want to check and cleanup if NewStringUTF fails. You only want to call SetObjectElementArray if NewStringUTF succeeds.

Reply:
Can `SetObjectElementArray` raise an exception?  
The index is within the array length and we store a string. I assume `cacheDirArray` is a string array.

Reply:
> ??? You want to check and cleanup if NewStringUTF fails. You only want to call SetObjectElementArray if NewStringUTF succeeds.

If the SetObjectArrayElement will throw an exception we will call the NewStringUTF while the exception is raised which is a kind of "jni" issue. If such an exception is possible we should check and cleanup.

Reply:
A quick search for `SetObjectArrayElement` in `java.desktop` module shows the call to `SetObjectArrayElement` is rarely followed by exception check.

What kind of exception can `SetObjectArrayElement` raise if the index is within the range and the stored element is of correct type?

Reply:
I think `NewStringUTF()` can throw OOM and also return `NULL`, just which one you pick.

Reply:
> I think `NewStringUTF()` can throw OOM and also return `NULL`, just which one you pick.

Yes.

But we're discussing whether a check for exception is necessary after `(*env)->SetObjectArrayElement`.

Reply:
That's exactly my point. Therefore the check for exception after `SetObjectArrayElement` is unnecessary.